cl/corpus_importer/management/commands/import_tn.py

```diff
@@ -0,0 +1,165 @@
+        "download_urls": case["pdf_url"],
+    }
+
+
+def import_tn_corpus(log, skip_until, filepath):
+    """Import TN Corpus
+
+    :param log: Should we view logging info
+    :param skip_until: Label ID, if any, to process first
+    :param filepath: Location of our overriding data json file
+    :return: None
+    """
+    ready = False if skip_until else True
+
+    if log:
+        logging.getLogger().setLevel(logging.INFO)
+
+    logging.info("Starting import")
+    tn_corpus = sorted(json.load(filepath), key=lambda x: x["label"])
+    if not ready:
+        case = [x for x in tn_corpus if x["label"] == skip_until][0]
+        logging.info(
+            "Skipping until case %s labeled: %s", case["title"], case["label"]
+        )
+    courts = {
+        "tennworkcompcl": Court.objects.get(pk="tennworkcompcl"),
+        "tennworkcompapp": Court.objects.get(pk="tennworkcompapp"),
+    }
+
+    for case in tn_corpus:
+        if case["label"] == skip_until:
+            ready = True
+        if not ready:
+            continue
+        logging.info(
+            "Processing label:%s for case:%s", case["label"], case["title"]
+        )
+        pdf_path = glob(
+            "%s/%s/*.pdf" % (os.path.dirname(filepath.name), case["label"])
+        )[0]
+        with open(pdf_path, "r") as p:
+            pdf_data = p.read()
+
+        sha1_hash = sha1(force_bytes(pdf_data))
+        ops = Opinion.objects.filter(sha1=sha1_hash)
+        if len(ops) > 0:
+            op = ops[0]
+            logging.warn(
+                "Document already in database. See: %s at %s"
+                % (op.get_absolute_url(), op.cluster.case_name)
+            )
+
+        docket, opinion, cluster, citations, error = make_objects(
+            make_item(case), courts[case["court"]], sha1_hash, pdf_data,
+        )
+
+        save_everything(
+            items={
+                "docket": docket,
+                "opinion": opinion,
+                "cluster": cluster,
+                "citations": citations,
+            },
+            index=False,
+        )
+
+        extract_doc_content.delay(
+            opinion.pk, do_ocr=True, citation_jitter=True,
+        )
+        logging.info(
+            "Successfully added Tennessee object cluster: %s", cluster.id
+        )
+
+
+class Command(VerboseCommand):
+    help = "Import TN data corpus received from TN Workers Comp boards."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--input-file",
+            type=file,
+            help="The filepath to our preprocessed data file.",
+            required=True,
+        )
+        parser.add_argument(
+            "--log",
+            action="store_true",
+            default=False,
+            help="Choose to view info log lines.",
+        )
+        parser.add_argument(
+            "--skip-until",
+            default=False,
+            help="Skip until to process",
+            type=int,
```

Comment:
Type `int`, default `False` is a bit weird. I'd default to zero or -1.

Reply:
thinking about it now - id probably just leave it with no default.  let it roll as none if its not addressed.